include/Nazara/Utility/Algorithm.hpp

```diff
@@ -8,17 +8,26 @@
 #define NAZARA_ALGORITHM_UTILITY_HPP
 
 #include <Nazara/Prerequesites.hpp>
+#include <Nazara/Core/Color.hpp>
 #include <Nazara/Core/SparsePtr.hpp>
 #include <Nazara/Math/Box.hpp>
 #include <Nazara/Math/Matrix4.hpp>
+#include <Nazara/Math/Quaternion.hpp>
 #include <Nazara/Math/Vector2.hpp>
 #include <Nazara/Math/Vector3.hpp>
+#include <Nazara/Math/Vector4.hpp>
 #include <Nazara/Utility/IndexIterator.hpp>
-#include <Nazara/Utility/Mesh.hpp>
-#include <Nazara/Utility/SkeletalMesh.hpp>
 
 namespace Nz
 {
+	class Joint;
+
+	class VertexStruct_XYZ_Normal_UV_Tangent_Skinning;
+	typedef VertexStruct_XYZ_Normal_UV_Tangent_Skinning SkeletalMeshVertex;
+
+	class VertexStruct_XYZ_Normal_UV_Tangent;
+	typedef VertexStruct_XYZ_Normal_UV_Tangent MeshVertex;
+
```

Comment:
Could you please put all the forward declaration together and put all of typedef (btw, using would be better) together too?